### ISSUE TITLE
esp32 hw test failed once with pulse counter/position mismatch

### ISSUE THREAD

**Issue by gin66** [gin66/FastAccelStepper#68]
One of the esp32 hw test sequences has failed once with getCurrentPosition() being off by 35 steps against the attached pulse counter, which is at 54us/step close to 2ms aka one queue command. Repeating the test has passed.

As the critical code is guarded with a pair of `noInterrupts()`/`interrupts()`, there is currently no good explanation.

**Comment by gin66** [comment]
Problem is here:
```
    noInterrupts();
    int32_t pos = queue_end.pos;
    uint8_t wp = next_write_idx;
    uint8_t rp = read_idx;
#if defined(ARDUINO_ARCH_ESP32)
    // pulse counter should go max up to 255 with perhaps few pulses overrun, so
    // this conversion is safe
    int16_t done_p = (int16_t)_getPerformedPulses();
#endif
    interrupts();
```
If between `noInterrupts()` and `_getPerformedPulses()` a step is generated, which causes the pulse counter to wrap around to zero, then the reading is off by a command‘s full step count. This risky time window is short <<1 us (assume 40ns at 160MHz cpu clock) every 2ms. So the stepper would run at high speed for average 100s to happen once.

Steps to solve this:
- build test case (let stepper run for e.g. 600s) and check for failure
- add code, which acts on `done_p == 0` and merely repeats this code sequence between `noInterrupts()` and `interrupts()`

**Comment by gin66** [comment]
Test seems to work, but the issue is bit more complex to solve. Still not ok.

**Comment by gin66** [comment]
Released version 0.23.2 with a fix
